### ISSUE TITLE
logs polluted with haproxy[17797]: ds: 0x7f20f406f660 when dns discovery is used

### ISSUE THREAD

**Issue by VigneshSP94** [haproxy/haproxy#1304]
## Detailed description of the problem

When using DNS discovery over TCP, haproxy logs gets filled with messages like `haproxy[17797]: ds: 0x7f20f406f660`

```
Jun 23 08:21:40 hostname haproxy[17797]: ds: 0x7f20f406f660
Jun 23 08:21:40 hostname haproxy[17797]: ds: 0x7f20f406f660
Jun 23 08:21:40 hostname haproxy[17797]: ds: 0x7f20f406f660
Jun 23 08:21:40 hostname haproxy[17797]: ds: 0x7f20f406f660
Jun 23 08:21:40 hostname haproxy[17797]: ds: 0x7f20f406f660
Jun 23 08:21:40 hostname haproxy[17797]: ds: 0x7f20f406f660
Jun 23 08:21:40 hostname haproxy[17797]: ds: 0x7f20f406f660
Jun 23 08:21:40 hostname haproxy[17797]: ds: 0x7f20f406f660
```
This is not affecting the functionality in anyway, since we are in the process of moving to the new version, we are trying to understand what is causing this.

## Expected behavior

Logs should not be polluted

## Steps to reproduce the behavior

1.  Configuring haproxy 2.4.1 with 50+ frontend and backends will trigger this.

## Do you have any idea what may have caused this?
No

## Do you have an idea how to solve the issue?
No

## What is your configuration?

```
global
  user haproxy
  group haproxy
  nbproc 1
  nbthread 16
  cpu-map auto:1/1-16 0-15
  log /dev/log local2
  log /dev/log local0 notice
  chroot /path
  pidfile /path/pid
  daemon
  master-worker
  maxconn 200000
  hard-stop-after 1h
  stats socket /path/to/stats mode 660 level admin expose-fd listeners
  tune.ssl.cachesize 3000000
  tune.ssl.lifetime 60000
  ssl-default-bind-ciphers ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256
  ssl-default-bind-options ssl-min-ver TLSv1.2 ssl-max-ver TLSv1.2
  server-state-file /tmp/haproxystate
  tune.bufsize 40960

defaults
  mode http
  log global
  retries 3
  timeout http-request 10s
  timeout queue 10s
  timeout connect 10s
  timeout client 1m
  timeout server 1m
  timeout tunnel 10m
  timeout client-fin 30s
  timeout server-fin 30s
  timeout check 10s
  option httplog
  option forwardfor except 127.0.0.0/8
  option redispatch
  load-server-state-from-file global

backend st_dst_global_http
        stick-table type ip size 100k expire 1s store http_req_rate(1s)
backend st_dst_global_tcp
        stick-table type ip size 100k expire 1s store conn_rate(1s)

resolvers mydns
    nameserver dns1 stream+ipv4@127.0.0.3:53
    accepted_payload_size 8192 # allow larger DNS payloads
    hold valid 10s


frontend load_balancer:80
        bind x.x.x.x:80 mss 1440 alpn h2,http/1.1
        mode http
        option httplog

        use_backend b::80::sg if { be_conn_free(80::sg) le 0 }
        default_backend 80::sg

backend 80::sg
        mode  http
        option httpchk
        http-check send meth GET uri /healthz ver HTTP/1.1 hdr Host some-header.com
        http-check expect status 200
        server-template sg 5 _servers._tcp.v2-sg._certtest.080f6da266bd9c800c9a89535e8cbfbc-80.backends.local resolvers mydns check port 80 maxconn 5000 maxqueue 1
        errorfile 503 /etc/static/busy-ja-html.http

backend b::80::sg
        mode http
        option httpchk
        http-check send meth GET uri /wwwcheck.html ver HTTP/1.1
        http-check expect status 200
        server busy-server unix@v2-busy-servers.sock enabled backup
        errorfile 503 /etc/static/busy-ja-html.http
```

## Output of `haproxy -vv` and `uname -a`

```
HAProxy version 2.4.0-6cbbecf 2021/05/14 - https://haproxy.org/
Status: long-term supported branch - will stop receiving fixes around Q2 2026.
Known bugs: http://www.haproxy.org/bugs/bugs-2.4.0.html
Running on: Linux 4.15.0-42-generic #45-Ubuntu SMP Thu Nov 15 19:32:57 UTC 2018 x86_64
Build options :
  TARGET  = linux-glibc
  CPU     = generic
  CC      = cc
  CFLAGS  = -O2 -g -Wall -Wextra -Wdeclaration-after-statement -fwrapv -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-clobbered -Wno-missing-field-initializers -Wtype-limits -Wshift-negative-value -Wshift-overflow=2 -Wduplicated-cond -Wnull-dereference
  OPTIONS = USE_PCRE=1 USE_LINUX_TPROXY=1 USE_LINUX_SPLICE=1 USE_LIBCRYPT=1 USE_OPENSSL=1 USE_ZLIB=1 USE_SYSTEMD=1 USE_OT=1 USE_PROMEX=1
  DEBUG   =

Feature list : +EPOLL -KQUEUE +NETFILTER +PCRE -PCRE_JIT -PCRE2 -PCRE2_JIT +POLL -PRIVATE_CACHE +THREAD -PTHREAD_PSHARED +BACKTRACE -STATIC_PCRE -STATIC_PCRE2 +TPROXY +LINUX_TPROXY +LINUX_SPLICE +LIBCRYPT +CRYPT_H +GETADDRINFO +OPENSSL -LUA +FUTEX +ACCEPT4 -CLOSEFROM +ZLIB -SLZ +CPU_AFFINITY +TFO +NS +DL +RT -DEVICEATLAS -51DEGREES -WURFL +SYSTEMD -OBSOLETE_LINKER +PRCTL +THREAD_DUMP -EVPORTS +OT -QUIC +PROMEX -MEMORY_PROFILING

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_THREADS=64, default=16).
Built with OpenSSL version : OpenSSL 1.1.1h  22 Sep 2020
Running on OpenSSL version : OpenSSL 1.1.1h  22 Sep 2020
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
Built with the Prometheus exporter as a service
Built with network namespace support.
Built with OpenTracing support.
Built with zlib version : 1.2.11
Running on zlib version : 1.2.11
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with PCRE version : 8.39 2016-06-14
Running on PCRE version : 8.39 2016-06-14
PCRE library supports JIT : no (USE_PCRE_JIT not set)
Encrypted password support via crypt(3): yes
Built with gcc compiler version 7.3.0

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
              h2 : mode=HTTP       side=FE|BE     mux=H2       flags=HTX|CLEAN_ABRT|HOL_RISK|NO_UPG
            fcgi : mode=HTTP       side=BE        mux=FCGI     flags=HTX|HOL_RISK|NO_UPG
       <default> : mode=HTTP       side=FE|BE     mux=H1       flags=HTX
              h1 : mode=HTTP       side=FE|BE     mux=H1       flags=HTX|NO_UPG
       <default> : mode=TCP        side=FE|BE     mux=PASS     flags=
            none : mode=TCP        side=FE|BE     mux=PASS     flags=NO_UPG

Available services : prometheus-exporter
Available filters :
	[  OT] opentracing
	[SPOE] spoe
	[CACHE] cache
	[FCGI] fcgi-app
	[COMP] compression
	[TRACE] trace

Linux hostname 4.15.0-42-generic #45-Ubuntu SMP Thu Nov 15 19:32:57 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by TimWolla** [comment]
This looks like forgotten debug code: https://github.com/haproxy/haproxy/blame/5ffb045ed12f14a0367bcb0aa39931806c84357a/src/dns.c#L173

**Comment by capflam** [comment]
Good catch @TimWolla, I'll remove it.

**Comment by capflam** [comment]
@TimWolla, except if you want to provide a patch. As you prefer.

**Comment by TimWolla** [comment]
> @TimWolla, except if you want to provide a patch. As you prefer.

Please do it yourself, no need to send patch emails to delete a single line from a file. It's also not rocket science, no need to put my name anywhere :grin:

**Comment by capflam** [comment]
copy that ! :)

**Comment by VigneshSP94** [comment]
@capflam sorry to comment in a closed ticket. May I ask if this fix is in a patch or will be available in a release?

**Comment by TimWolla** [comment]
The fix will be included in HAProxy 2.4.2.
